From 12142a6cf00fda20df18afac4d4011ab49748c0d Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 15 Jan 2023 14:11:34 +0200
Subject: [PATCH] [SW-66343] gaudi2: fw_cfg_skip: remove flag

fw_cfg_skip feature was added in gaudi2 as part of
poweron, and is for a long time not maintained nor functional.
Hence, this patch removes this dead code, and as a result
it also frees space.

Change-Id: I2b67ebc575351e447ead1ae54e5ba03722f2210c
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 70f49c2..d5263d3 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -50,7 +50,6 @@
 
 static struct hl_rsp_err pending_error;
 static u64 preboot_en_caps;
-static u8 fw_cfg_skip;
 static u8 eeprom_def_boot_en;
 
 static struct cmd_module_ops cmd_ops[] = {
@@ -518,9 +517,6 @@
 	preboot_version();
 	fuse_version();
 	preboot_read_straps();
-	if (hl_skip_fw_cfg())
-		preboot_puts("FW_CFG_SKIP: enabled\n");
-
 	preboot_read_test_mode();
 	preboot_security_dynamic_vers_init(ver_off, sizeof(preboot_vers));
 
@@ -581,11 +577,6 @@
 	return 1;
 }
 
-void preboot_set_fw_cfg_skip(u8 skip)
-{
-	fw_cfg_skip = skip;
-}
-
 void preboot_set_eeprom_defaults_en(u8 eeprom_defaults_on_err)
 {
 	eeprom_def_boot_en = eeprom_defaults_on_err;
@@ -600,9 +591,6 @@
 
 void preboot_finalize_uboot_override(void)
 {
-	if (fw_cfg_skip)
-		boot_info_set(HL_BOOT_INFO_FW_CFG_SKIP, &fw_cfg_skip, sizeof(fw_cfg_skip));
-
 	if (eeprom_def_boot_en)
 		preboot_boot_info_set_eeprom_fw_info();
 }
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 2777498..e9df8c6 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -214,11 +214,6 @@
 		.use_calib = false, .calib_offset = 8, .hl_tmp_data = { 0 }}
 };
 
-bool hl_fw_cfg_skip_is_supported(void)
-{
-	return true;
-}
-
 void hl_ts_data_init(void)
 {
 	u32 pll_freq;
diff --git a/include/preboot.h b/include/preboot.h
index ec0c517..508ff3e 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -245,11 +245,9 @@
 void preboot_init_pcie_phy(void);
 void preboot_engines_init(void);
 void preboot_clear_wd_gpio(void);
-void preboot_set_fw_cfg_skip(u8 skip);
 void preboot_set_eeprom_defaults_en(u8 eeprom_defaults_on_err);
 void preboot_finalize_uboot_override(void);
 void preboot_update_bar_sizes(void);
-bool hl_fw_cfg_skip_is_supported(void);
 
 void ppboot_init_stack_mon_dev_specific(void);
 void ppboot_check_stack_mon_dev_specific(void);
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 73cabd9..8a569ae 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -99,25 +99,6 @@
 	return err;
 }
 
-__weak bool hl_fw_cfg_skip_is_supported(void) { return false; }
-
-static int module_lkd_msg_fw_cfg_skip(struct lkd_fw_comms_msg *mcomms)
-{
-	if (!hl_fw_cfg_skip_is_supported()) {
-		preboot_puts("fw_cfg_skip is not supported\n");
-		return -1;
-	}
-
-	if (preboot_security_is_enabled()) {
-		preboot_puts("fw_cfg_skip is not supported due to security\n");
-		return -1;
-	}
-
-	preboot_set_fw_cfg_skip(mcomms->fw_cfg_skip);
-
-	return 0;
-}
-
 static int module_lkd_msg_eeprom_defaults(struct lkd_fw_comms_msg *mcomms)
 {
 	if (preboot_security_is_enabled()) {
@@ -138,9 +119,6 @@
 	switch (mcomms->header.type) {
 	case HL_COMMS_RESET_CAUSE_TYPE:
 		err = module_lkd_msg_reset_cause(mcomms);
-		break;
-	case HL_COMMS_FW_CFG_SKIP_TYPE:
-		err = module_lkd_msg_fw_cfg_skip(mcomms);
 		break;
 	case HL_COMMS_EEPROM_DEFAULTS_TYPE:
 		err = module_lkd_msg_eeprom_defaults(mcomms);
